ctapipe/ml/tools/apply.py

```diff
@@ -0,0 +1,220 @@
+            "Don't overwrite tables in output file if it exists",
+        ),
+        "f": (
+            {"Apply": {"overwrite": True}},
+            "Overwrite output file if it exists",
+        ),
+    }
+
+    classes = [
+        TableLoader,
+        EnergyRegressor,
+        ParticleIdClassifier,
+        StereoCombiner,
+    ]
+
+    def setup(self):
+        """"""
+        self.log.info("Copying to output destination.")
+        shutil.copy(self.input_url, self.output_path)
+
+        self.h5file = tables.open_file(self.output_path, mode="r+")
+        self.loader = TableLoader(
+            parent=self,
+            h5file=self.h5file,
+            load_dl1_images=False,
+            load_dl1_parameters=True,
+            load_dl2=True,
+            load_simulated=True,
+            load_instrument=True,
+        )
+
+        self.apply_regressor = self._setup_regressor()
+        self.apply_classifier = self._setup_classifier()
+
+    def _setup_regressor(self):
+        if self.energy_regressor_path is not None:
+            self.regressor = EnergyRegressor.read(
+                self.energy_regressor_path,
+                parent=self,
+            )
+            self.regressor_combine = StereoCombiner.from_name(
+                self.stereo_combiner_type,
+                combine_property="energy",
+                algorithm=self.regressor.model_cls,
+                parent=self,
+            )
+            return True
+        return False
+
+    def _setup_classifier(self):
+        if self.particle_classifier_path is not None:
+            self.classifier = ParticleIdClassifier.read(
+                self.particle_classifier_path,
+                parent=self,
+            )
+            self.classifier_combine = StereoCombiner.from_name(
+                self.stereo_combiner_type,
+                combine_property="classification",
+                algorithm=self.classifier.model_cls,
+                parent=self,
+            )
+            return True
+        return False
+
+    def start(self):
+        if self.apply_regressor:
+            self.log.info("Apply regressor.")
+            mono_predictions = self._apply(self.regressor, "energy")
+            self._combine(self.regressor_combine, mono_predictions)
+
+        if self.apply_classifier:
+            self.log.info("Apply classifier.")
+            mono_predictions = self._apply(self.classifier, "classification")
+            self._combine(self.classifier_combine, mono_predictions)
+
+    def _apply(self, reconstructor, parameter):
+        prefix = reconstructor.model_cls
+
+        tel_tables = []
+
+        desc = f"Applying {reconstructor.__class__.__name__}"
+        unit = "telescope"
+        for tel_id, tel in tqdm(self.loader.subarray.tel.items(), desc=desc, unit=unit):
+            if tel not in reconstructor._models:
+                self.log.warning(
+                    "No regressor model for telescope type %s, skipping tel %d",
```

Comment:
This warning is also used for the particle classifier, so I would cut the "regressor".

Reply:
done